coderd/httpmw/apikey.go

```diff
@@ -149,9 +150,23 @@ func ExtractAPIKey(db database.Store, oauth *OAuth2Configs, redirectToLogin bool
 			// Tracks if the API key has properties updated!
 			changed := false
 
-			if key.LoginType != database.LoginTypePassword {
+			var link database.UserLink
+			// The login_type should never be empty but sometimes it is
+			// for tests.
+			if key.LoginType != "" && key.LoginType != database.LoginTypePassword {
+				link, err = db.GetUserLinkByUserIDLoginType(r.Context(), database.GetUserLinkByUserIDLoginTypeParams{
```

Comment:
It does kinda suck that we have to do two DB queries on each API request now no matter what... but I suppose there isn't a _great_ way to avoid such a thing.